### PR TITLE
Qemu 12 nodes in 4 different clouds

### PR DESCRIPTION
**Pull request by altergui**
started working in a virtualized mesh cloud

### CODE REVIEW

**Comment by spiccinini** [comment]
Looks good! I will test it :)

File: tools/qemu_cloud_start [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+for id in `seq 0 10` ; do
+  #./tools/qemu_dev_start --no-serial --node-id $id $@ &
```
Comment (spiccinini) [inline tools/qemu_cloud_start:3]: This is commented on purpose?

**Comment by spiccinini** [comment]
What is the idea behind the ETH2 interface?

File: tools/qemu_cloud_start (hunk 1 above)
Comment (altergui) [reply tools/qemu_cloud_start]: nope. sorry about that, anyway refactored everything into an ansible-playbook, much more powerful

**Comment by altergui** [comment]
after major work, made an ansible playbook that reads a YAML describing a complex set of nodes distributed accross different clouds with links connecting them.

**Review by G10h4ck — APPROVED** [review]

File: tools/qemu_dev_start [hunk 2]
```diff
@@ -62,9 +65,12 @@ print_help()
 	printf '\t%s\n' "<rootfs>: Path to a x86-64-generic-rootfs.tar.gz"
```
Comment (spiccinini) [inline tools/qemu_dev_start:65]: The usage line (--help) has to be updated also.  
We have been using argbash https://argbash.io/generate to maintain the options of the script (and helper line, etc), can you please use it? (you can copy from the line 0 until # ARGBASH_GO() into the argbash textbox and then copy the result back to the script).

Comment (altergui) [reply tools/qemu_dev_start:66]: thanks! was unaware of that. rebased and force-pushed my branch, using argbash

**Comment by altergui** [comment]
with this rebase i dropped the `--no-serial` argument (it's not really important) and used argbash to handle arguments, as indicated
https://github.com/libremesh/lime-packages/compare/1143e39753d7a43953366e17ca79254223db1bc9..5407071393ed5c3868d377e0973e2b4d3b502804

**Comment by spiccinini** [comment]
I tested the non cloud operation and it is working :)

At first it the `--enable-wan` options wasn't working and It took me some time to discover that as the MAC configuration  changed the dnsmasq lease time was preventing dnsmasq to offer an ip as the range is only one IP .... Anyway, everything good, I've just deleted the lease file (`/var/lib/misc/dnsmasq.leases` in my distro).

**Review by spiccinini — APPROVED** [review]

**Comment by spiccinini** [comment]
One thing before merging. Which license the python files have?

File: tools/qemu_cloud_start [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```
Comment (spiccinini) [inline tools/qemu_cloud_start:1]: This and qemu_cloud_stop are no longer needed?

**Comment by spiccinini** [comment]
@altergui please can you answer the remaining questions? I want to merge this :)

**Comment by altergui** [comment]
> One thing before merging. Which copyright and license the python files have?

good catch! no license whatsoever. wrote an email to the author asking for clarification.
i think  (based on other code by same author) it's safe to assume the author intended to publish under MIT license and just ommitted it.
we can wait a bit until a proper author statement, and if that doesn't happen... no idea

File: tools/qemu_cloud_start (hunk 3 above)
Comment (altergui) [reply tools/qemu_cloud_start]: true. i just removed them

**Comment by altergui** [comment]
success. MIT license it is.
https://gist.github.com/upa/94f82e00dfffbb25d896/revisions

**Comment by spiccinini** [comment]
Thank gui!

**Pull request merged**